04-cond.md

```diff
@@ -0,0 +1,348 @@
+    end
+end
+
+disp(['sum of positive values: ', num2str(total)])
+~~~
+
+
+~~~ {.output}
+sum of positive values: 20
+~~~
+
+With a little extra effort, we can calculate the
+positive and negative sums in a loop:
+
+~~~ {.matlab}
+pos_total = 0;
+neg_total = 0;
+
+for n = numbers
+    if n >= 0
+        pos_total = total + n;
+    else
+        neg_total = neg_total + n;
+    end
+end
+
+disp(['sum of positive values: ', num2str(pos_total)]);
+disp(['sum of negative values: ', num2str(neg_total)]);
+~~~
+
+~~~ {.output}
+sum of positive values: 26
+sum of negative values: -6
+~~~
+
+We can even put one loop inside another:
+
+~~~ {.matlab}
+for consonant = 'bcd'
+    for vowel = 'ae'
+        disp (strcat(consonant, vowel));
+    end
+end
+~~~
+
+~~~ {.output}
+ba
+be
+ca
+ce
+da
+de
+~~~
+
+> ## Nesting {.challenge}
+>
+> 1. Will changing the order of nesting in the above loop change
+> the output? Why? Write down the output you might expect from
+> changing the order of the loops, then rewrite the code to test
+> your hypothesis.
+>
+> 2. MATLAB (and most other languges in the C family) provides
+> [in-place operators](reference.html#in-place-operator) that
+> work like this:
+>
+> 
+> ~~~{.matlab}
+> x = 1;
+> x += 1;
+> x *= 3;
+> ~~~
+>
+> Rewrite the code that sums the positive and negative values
+> in an array using these in-place operators. Do you think that
+> the result is more or less readable than the original?
+
+Currently, our script `analyze.m` reads in data, analyzes it,
+and saves plots of the results.
+If we would rather display the plots interactively,
+we would have to remove (or *comment out*) the following code:
+
+~~~{.matlab}
+print('-dpng', img_name);
+close();
+~~~
+
+And, we'd also have to change this line of code, from:
+
+~~~{.matlab}
+figure('visible', 'off')
+~~~
+
+to:
+
+~~~{.matlab}
+figure('visible', 'on')
+% or equivalently: figure()
+~~~
+
+This is not a lot of code to change every time,
+but it's still work that's easily avoided using conditionals.
+Here's our script re-written to use *conditionals*
+to switch between saving plots as images and plotting them interactively:
+
+~~~{.matlab}
+
+% plot_switch:
+%   0 - show plots interactively
+%   1 - save plots to disk
+
+plot_switch = 0;
+
+for idx = 1:3
+
+    % Generate strings for file and image names:
+    file_name = sprintf('data/inflammation-%02d.csv', idx);
+    img_name = sprintf ('patient_data-%02d.png', idx);
+
+    patient_data = csvread(file_name);
+    ave_inflammation = mean(patient_data, 1);
+
+    if plot_switch == 1
+        figure('visible', 'off')
+    else
+        figure('visible', 'on')
+    end
+
+    subplot(2, 2, 1);
+    plot(ave_inflammation);
+    ylabel('average')
+
+    subplot(2, 2, 2);
+    plot(max(patient_data, [], 1));
+    ylabel('max')
+
+    subplot(2, 2, 3);
+    plot(min(patient_data, [], 1));
+    ylabel('min')
+
+    if plot_switch == 1
+        print('-dpng', img_name);
+        close()
+    end
+
+end
+~~~
```

Comment:
I sometimes add a challenge and ask people to have two different switches, one for printing, one for visibility.
That might not be the cleverest challenge, but I think we might need one at the end of this section..